Fix early morning preference for service booking

fawkesley opened the pull request:
First, a little refactor: factor out `getServiceFromSlug` as it's repeated code

Then enable the early morning appointments page for service appointments (by adding the `service_slug` URL parameter in `routes.js`) and make the following resulting fixes:
- Tweak the appointment times so it actually makes sense when you click early appointments.
- Actually pass in `address` for each appointment in every page, not just `next-available-appointment`
- Actually hide filters for the early morning appointments page according to the service filters

![image](https://cloud.githubusercontent.com/assets/254290/11270859/3691a134-8eba-11e5-8867-f9a175b4fcaf.png)

_click early morning appointment_

![image](https://cloud.githubusercontent.com/assets/254290/11270939/d1cdec02-8eba-11e5-92a7-62ada9c8251e.png)

Commit:
Tweak service appointment times for early morning

Ensure that for each service there's an appointment and an early morning
appointment at a later date.

Commit:
Factor out getServiceFromSlug function

It's repeated code, and we're about to use it again.

Commit:
Support early morning appointments for services

Commit:
Hide preference filters on early morning page

As in 423eddf7 but for all the booking preference pages.

This is getting unwieldy and ultimately we should definitely be doing
this in code. But now doesn't seem like the right time!

Commit:
Fix missing address context on appointment pages

This meant the address was lost for the diabetes eye screening
appointments when selecting early morning appointments.

paulfurley commented:
Ah, I've just noticed the practice address is showing up when it shouldn't be. Closing until it's fixed.

Commit:
Fix the `practice` context passed in to appts page

We should be passing in the local `practice` variable rather than first
one in the list of practices.

evilstreak commented:
:+1: Thanks for the super clear commits
